components/raftstore/src/store/fsm/store.rs

```diff
@@ -1067,16 +1067,38 @@ impl RaftBatchSystem {
             future_poller: workers.future_poller.sender().clone(),
         };
         let region_peers = builder.init()?;
-        self.start_system(workers, region_peers, builder, dyn_cfg)?;
+        let engine = RocksEngine::from_db(builder.engines.kv.clone());
+        if engine.support_write_batch_vec() {
+            self.start_system(
+                workers,
+                region_peers,
+                builder,
+                dyn_cfg,
+                engine.write_batch_vec(0, 0),
+            )?;
+        } else {
+            self.start_system(
+                workers,
+                region_peers,
+                builder,
+                dyn_cfg,
+                engine.write_batch(),
+            )?;
+        }
         Ok(())
     }
 
-    fn start_system<T: Transport + 'static, C: PdClient + ConfigClient + 'static>(
+    fn start_system<
+        T: Transport + 'static,
+        C: PdClient + ConfigClient + 'static,
+        W: WriteBatch + WriteBatchVecExt<RocksEngine> + 'static,
+    >(
         &mut self,
         mut workers: Workers,
         region_peers: Vec<SenderFsmPair<RocksEngine>>,
         builder: RaftPollerBuilder<T, C>,
         dyn_cfg: Box<dyn DynamicConfig>,
+        phantom: W,
```

Comment:
Should give it a meaningful name.

Reply:
It is just a phantom object, to make this struct can implement different traits.